src/lib/support/CHIPMem.cpp

```diff
@@ -36,31 +37,51 @@ namespace Platform {
 extern CHIP_ERROR MemoryAllocatorInit(void * buf, size_t bufSize);
 extern void MemoryAllocatorShutdown();
 
-static std::atomic_int memoryInitializationCount{ 0 };
+#if !CHIP_SYSTEM_CONFIG_NO_LOCKING
+static chip::System::Mutex gMemoryLock;
+#endif
+static int memoryInitializationCount{ 0 };
```

Comment:
If CHIP_SYSTEM_CONFIG_NO_LOCKING is enabled, I am unsure if atomic variables can exist (since no locking mechanisms are supported by the platform presumably)...
If CHIP_SYSTEM_CONFIG_NO_LOCKING is not enabled, there shouldn't be any issues...

Reply:
> If CHIP_SYSTEM_CONFIG_NO_LOCKING is enabled

This is actually the context of the original issue, fwiw: it's reported on a platform where CHIP_SYSTEM_CONFIG_NO_LOCKING  is true....

Reply:
Got it!

Reply:
@bzbarsky-apple please feel free to comment or approve/merge. Thanks!